gateway/mock_response.go

```diff
@@ -0,0 +1,50 @@
+package gateway
+
+import (
+	"bytes"
+	"io"
+	"net/http"
+
+	"github.com/TykTechnologies/tyk/apidef/oas"
+	header "github.com/TykTechnologies/tyk/headers"
+	"github.com/getkin/kin-openapi/routers"
+)
+
+func (p *ReverseProxy) mockResponse(r *http.Request, route *routers.Route) (res *http.Response, err error) {
+	res = &http.Response{Header: http.Header{}}
+
+	var code int
+	var contentType string
+	var body []byte
+	var headers map[string]string
+
+	tykMockRespOp := p.TykAPISpec.OAS.GetTykExtension().Middleware.Operations[route.Operation.OperationID].MockResponse
```

Comment:
Should it actually check if map record exists or not?

Reply:
It has the check in the outer function so here, it is safe to access to the map.

Reply:
This is just a read operation. When a new update comes, it is not updating one field, a fresh api is loaded from scratch. And the global apis variable is protected with mutexes.